src/displaysize.rs

```diff
@@ -1,31 +1,82 @@
 //! Display size
 
-// TODO: Add to prelude
-/// Display size enumeration
-#[derive(Clone, Copy)]
-pub enum DisplaySize {
-    /// 128 by 64 pixels
-    Display128x64,
-    /// 128 by 32 pixels
-    Display128x32,
-    /// 96 by 16 pixels
-    Display96x16,
-    /// 70 by 42 pixels
-    Display72x40,
-    /// 64 by 48 pixels
-    Display64x48,
+use super::command::Command;
+use generic_array::ArrayLength;
+use typenum::{
+    Unsigned, U0, U1024, U128, U16, U192, U28, U32, U360, U384, U40, U48, U512, U64, U72, U96,
+};
+
+pub trait DisplaySize {
+    type Width: Unsigned;
+    type Height: Unsigned;
+    type OffsetX: Unsigned;
+    type OffsetY: Unsigned;
+    type BufferSize: ArrayLength<u8>;
+
+    fn ComPinConfig() -> Command;
+}
+
+pub struct DisplaySize128x64;
+impl DisplaySize for DisplaySize128x64 {
+    type Width = U128;
+    type Height = U64;
+    type OffsetX = U0;
+    type OffsetY = U0;
```

Comment:
With `const` changes suggested above:

```suggestion
    const WIDTH: u8 = 128;
    const HEIGHT: u8 = 64;
```